src/main/java/Parser.java

```diff
@@ -0,0 +1,324 @@
+        assert stringLength >= 1 : "Input cannot be empty";
+        String stringIndex = input.substring(5, input.length());
+        int index = Integer.parseInt(stringIndex);
+        Task chosen = this.tasks.get(index - 1);
+        chosen.finish();
+        storage.saveListToData(this.tasks.get());
+        return ui.markAsDoneMsg(chosen);
+    }
+
+    /**
+     * Checks the type of Task the user is trying to create
+     *
+     * @param input is the User's input in String form
+     * @throws EmptyDescriptionException
+     * @throws WrongFormatException
+     * @return a String Message
+     */
+    private String newTaskEntry(String input) throws EmptyDescriptionException, WrongFormatException {
+        int stringLength = input.length();
+        assert stringLength >= 1 : "Input cannot be empty";
+        if (input.contains("todo")) {
+            return this.createAndAddTodo(input);
+        } else if (input.contains("deadline")) {
+            return this.createAndAddDeadline(input);
+        } else {
+            return this.createAndAddEvent(input);
+        }
+    }
+
+    /**
+     * Creates and adds a Task with type "Todo" to the TaskList
+     *
+     * @param input is the description of the task as given by user
+     * @throws EmptyDescriptionException
+     * @throws WrongFormatException
+     * @return a String Message
+     */
+    private String createAndAddTodo(String input) throws EmptyDescriptionException, WrongFormatException {
+        int stringLength = input.length();
+        assert stringLength >= 1 : "Input cannot be empty";
+        if (input.length() < 5 || input.substring(5).replaceAll("\\s", "").equals("")) {
+            return new EmptyDescriptionException("todo").toString();
+        } else if (!Character.toString(input.charAt(4)).equals(" ")) {
+            return new WrongFormatException("todo").toString();
+        } else {
+            Task task = new Todo(input.substring(5, input.length()));
+            return this.addTaskToTasklist(task);
+        }
+    }
+
+    /**
+     * Creates and adds a Task with type "Deadline" to the TaskList
+     *
+     * @param input is the description of the task as given by user
+     * @throws EmptyDescriptionException
+     * @throws WrongFormatException
+     * @return a String Message
+     */
+    private String createAndAddDeadline(String input) throws EmptyDescriptionException, WrongFormatException {
+        int stringLength = input.length();
+        assert stringLength >= 1 : "Input cannot be empty";
+        if (input.length() < 9 || input.substring(8).replaceAll("\\s", "").equals("")) {
+            throw new EmptyDescriptionException("deadline");
+        } else if (input.contains("/by")
+                && Character.toString(input.charAt(8)).equals(" ")
+                && Character.toString(input.charAt(input.indexOf("/") + 3)).equals(" ")
+                && Character.toString(input.charAt(input.indexOf("/") - 1)).equals(" ")) {
```

Comment:
The spacing here makes the code readable but I feel that this might be a complicated expression that is hard to follow through without additional context.